src/cpp/fastdds/publisher/DataWriterImpl.hpp

```diff
@@ -662,17 +662,15 @@ class DataWriterImpl : protected rtps::IReaderDataFilter
             const fastdds::rtps::WriterAttributes& writer_attributes,
             bool& is_datasharing_compatible) const;
 
-    template<typename SizeFunctor>
     bool get_free_payload_from_pool(
```

Comment:
Now that this is not a template, we can move the implementation to the source file

Reply:
Done in 3984949b2